### PR TITLE
feat: add terraform_readme (OPS-3367)

### PR DESCRIPTION
**Pull request by GabeL7r**

### CODE REVIEW

File: terraform_readme.sh [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  let "index+=1"
+done
+
+for path_uniq in $(echo "${paths[*]}" | tr ' ' '\n' | sort -u); do
+  path_uniq="${path_uniq//__REPLACED__SPACE__/ }"
+
+  pushd "$path_uniq" > /dev/null
+  if [[ -f README.md ]]; then
+    if grep -q "BEGINNING OF TERRAFORM-DOCS HOOK" README.md; then
+      DOCS=$(terraform-config-inspect ./)
```
Comment (GabeL7r) [inline terraform_readme.sh:27]: change between this and terraform docs is replacing terraform-docs md with terraform-config-inspect

File: .pre-commit-hooks.yaml [hunk 2]
```diff
@@ -14,6 +14,14 @@
   files: (\.tf)$
   exclude: \.terraform\/.*$
 
+- id: terraform_readme
+  name: Terraform Readme
+  description: Creates readme for terraform modules.
```
Comment (mrwacky42) [inline .pre-commit-hooks.yaml]:
```suggestion
  description: Creates a README for Terraform modules using `terraform_config_inspect`.
```

File: hooks.yaml [hunk 3]
```diff
@@ -14,6 +14,14 @@
   files: (\.tf)$
   exclude: \.terraform\/.*$
 
+- id: terraform_readme
+  name: Terraform Readme
+  description: Creates readme for terraform modules.
```
Comment (mrwacky42) [inline hooks.yaml]:
```suggestion
  description: Creates a README for Terraform modules using `terraform_config_inspect`.
```

**Review by mrwacky42 — COMMENTED** [review]
While we are here, can we fix the 2 pass problem?
When it creates an empty one, it is just:
```
<begin_hook>
<end_hook>
```
Then the second pass adds a blank line:
```
<begin_hook>

<end_hook>
```

**Comment by mrwacky42** [comment]
Link to a Jira please

File: terraform_readme.sh [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+      # Assume the entire file is autogenerated. Replace the contents with the new generated docs and add the hook so future changes won't hit this code path
+      echo "<!-- BEGINNING OF TERRAFORM-DOCS HOOK -->" > README.md
+      terraform-config-inspect ./ >> README.md
+      echo "<!-- END OF TERRAFORM-DOCS HOOK -->" >> README.md
+      echo "Updating README.md, please git add."
+      exit 1
+    fi
+  else
+    echo "<!-- BEGINNING OF TERRAFORM-DOCS HOOK -->" > README.md
+    terraform-config-inspect ./ >> README.md
```
Comment (GabeL7r) [inline terraform_readme.sh:39]: @mrwacky42 i think removing sed addresses the inconsistency

**Review by mrwacky42 — APPROVED** [review]

**Pull request merged**